File: src/main/java/me/centralhardware/znatoki/telegram/statistic/entity/TelegramUser.java
package me.centralhardware.znatoki.telegram.statistic.entity;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.Objects;
import java.util.UUID;

@Builder
@Getter
@Setter
public final class TelegramUser {
    private Long id;
    private Role role;
    private UUID organizationId;
    private List<Long> services;

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/entity/TelegramUser.kt
package me.centralhardware.znatoki.telegram.statistic.entity

import kotliquery.Row
import me.centralhardware.znatoki.telegram.statistic.parseLongList

class TelegramUser(
    val id: Long,
    val permissions: List<Permissions>,
    val services: List<Long>,
    val name: String,
)

fun Row.parseUser() =
    TelegramUser(
        long("id"),
        array<String>("permissions").map { Permissions.valueOf(it) },
        string("services").parseLongList(),
        string("name"),
    )
